### PR TITLE
Signal pipe registration

### PR DESCRIPTION
**Pull request by babrauskas**
From example demo_https.c  server. Server can often crash without pipe signal ignore registration.
https://stackoverflow.com/questions/108183/how-to-prevent-sigpipes-or-handle-them-properly

Additional info from microhttp docs:
section SIGPIPE
cindex signals
MHD does not install a signal handler for SIGPIPE.  On platforms
where this is possible (such as GNU/Linux), it disables SIGPIPE for
its I/O operations (by passing MSG_NOSIGNAL).  On other platforms,
SIGPIPE signals may be generated from network operations by
MHD and will cause the process to die unless the developer
explicitly installs a signal handler for SIGPIPE.

**Hence portable code using MHD must install a SIGPIPE handler or
explicitly block the SIGPIPE signal.**  MHD does not do so in order
to avoid messing with other parts of the application that may
need to handle SIGPIPE in a particular way.  You can make your application handle SIGPIPE by calling the following function in main():

static void
catcher (int sig)
{
}

static void
ignore_sigpipe ()
{
  struct sigaction oldsig;
  struct sigaction sig;

  sig.sa_handler = &catcher;
  sigemptyset (&sig.sa_mask);
#ifdef SA_INTERRUPT
  sig.sa_flags = SA_INTERRUPT;  /* SunOS */
#else
  sig.sa_flags = SA_RESTART;
#endif
  if (0 != sigaction (SIGPIPE, &sig, &oldsig))
    fprintf (stderr,
             "Failed to install SIGPIPE handler: %s\n", strerror (errno));
}

### CODE REVIEW

File: examples/rest-server/restserver.c [hunk 1]
```diff
@@ -41,6 +41,41 @@ static void sigint_handler(int signo)
...
+/**
+ * setup handlers to ignore SIGPIPE.
+ */
+#ifndef MINGW
+static void
+ignore_sigpipe ()
+{
+  struct sigaction oldsig;
+  struct sigaction sig;
+
```
Comment (valinskas) [inline examples/rest-server/restserver.c]: memset(&sig, 0, sizeof(sig));

File: examples/rest-server/restserver.c [hunk 2]
```diff
@@ -38,9 +38,50 @@
 static volatile int restserver_quit;
 static void sigint_handler(int signo)
 {
+    puts("SIGINT occurs.\n");
```
Comment (GiedriusM) [inline examples/rest-server/restserver.c]: Nereikia debug printo, nes čia ne klaida, o tiesiog terminavimo signalas.

File: examples/rest-server/restserver.c [hunk 3]
```diff
@@ -38,9 +38,50 @@
...
+/**
+ * Function called if we get a SIGPIPE. Does counting.
+ * exmp. killall -13  restserver
+ * @param sig will be SIGPIPE (ignored)
+ */
+static void catcher(int sig)
+{
+    static volatile int sigpipe_cnt;
+    sigpipe_cnt++;
+    fprintf(stderr, "SIGPIPE occurs: %d times.\n",sigpipe_cnt); /* do nothing */
```
Comment (GiedriusM) [inline examples/rest-server/restserver.c]: stilius: kokia to komentaro gale prasmė? fprintf tikrai kažką daro.

File: examples/rest-server/restserver.c [hunk 4]
```diff
@@ -38,9 +38,50 @@
...
+    puts("SIGINT occurs.\n");
     restserver_quit = 1;
 }
 
+/**
+ * Function called if we get a SIGPIPE. Does counting.
+ * exmp. killall -13  restserver
+ * @param sig will be SIGPIPE (ignored)
+ */
+static void catcher(int sig)
```
Comment (GiedriusM) [inline examples/rest-server/restserver.c]: stilius: pirma `catcher` yra bendrinis pavadinimas, antra panašios funkcijos turėtų turėti panašius pavadinimus. jei SIGINT yra handlinama su `sigint_handler` tai ir SIGPIPE turėtų būti `sigpipe_handler`.

File: examples/rest-server/restserver.c [hunk 5]
```diff
@@ -38,9 +38,50 @@
...
+{
+    static volatile int sigpipe_cnt;
+    sigpipe_cnt++;
+    fprintf(stderr, "SIGPIPE occurs: %d times.\n",sigpipe_cnt); /* do nothing */
+}
+
+
+/**
+ * setup handlers to ignore SIGPIPE.
```
Comment (GiedriusM) [inline examples/rest-server/restserver.c]: Šita funkcija ne tik SIGPIPE sukonfigūruoja.

File: examples/rest-server/restserver.c [hunk 6]
```diff
@@ -38,9 +38,50 @@
...
+ */
+static void init_signals(void)
+{
+    struct sigaction oldsig;
+    struct sigaction sig;
+
+    //signal(SIGINT, sigint_handler);//automaticaly do SA_RESTART, we must break system functions exmp. select
```
Comment (GiedriusM) [inline examples/rest-server/restserver.c]: Kodo komentaruose neturėtų būti

File: examples/shared/commandline.c [hunk 7]
```diff
@@ -190,7 +190,7 @@ void output_buffer(FILE * stream,
         int j;
 
         print_indent(stream, indent);
-        memcpy(array, buffer+i, 16);
```
Comment (GiedriusM) [inline examples/shared/commandline.c]: Šitas pakeitimas nepriklauso SIGPIPE apdorojimui.

**Pull request merged**